service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/FurtherEvidenceUploadedEventHandlerTest.java

```diff
@@ -0,0 +1,320 @@
+        CaseData caseData = buildCaseDataWithNonConfidentialDocuments(HMCTS_USER);
+        FurtherEvidenceUploadedEvent furtherEvidenceUploadedEvent =
+            new FurtherEvidenceUploadedEvent(
+                caseData,
+                buildCaseDataWithConfidentialDocuments(HMCTS_USER),
+                false,
+                userDetailsHMCTS()
+            );
+
+        when(furtherEvidenceNotificationService.getRespondentRepresentativeEmails(caseData)).thenReturn(
+            representativeSolicitors);
+        when(furtherEvidenceNotificationService.getLocalAuthoritySolicitorEmails(caseData)).thenReturn(laSolicitors);
+
+        furtherEvidenceUploadedEventHandler.handleDocumentUploadedEvent(furtherEvidenceUploadedEvent);
+
+        verify(furtherEvidenceNotificationService).sendFurtherEvidenceDocumentsUploadedNotification(
+            caseData, representativeSolicitors, SENDER);
+        verify(furtherEvidenceNotificationService).sendFurtherEvidenceDocumentsUploadedNotification(
+            caseData, laSolicitors, SENDER);
+    }
+
+    @Test
+    void shouldNotSendNotificationWhenConfidentialDocIsUploadedByHMCTS() {
+        CaseData caseData = buildCaseDataWithConfidentialDocuments(HMCTS_USER);
+        FurtherEvidenceUploadedEvent furtherEvidenceUploadedEvent =
+            new FurtherEvidenceUploadedEvent(
+                caseData,
+                buildCaseDataWithNonConfidentialDocuments(HMCTS_USER),
+                false,
+                userDetailsHMCTS()
+            );
+
+        furtherEvidenceUploadedEventHandler.handleDocumentUploadedEvent(furtherEvidenceUploadedEvent);
+
+        verify(furtherEvidenceNotificationService, never()).sendFurtherEvidenceDocumentsUploadedNotification(
+            any(), any(), any());
+        verify(furtherEvidenceNotificationService, never()).sendFurtherEvidenceDocumentsUploadedNotification(
+            any(), any(), any());
+    }
+
+    @Test
+    void shouldSendNotificationWhenNonConfidentialDocIsUploadedByRespSolicitor() {
+        CaseData caseData = buildCaseDataWithNonConfidentialDocuments(REP_USER);
+        FurtherEvidenceUploadedEvent furtherEvidenceUploadedEvent =
+            new FurtherEvidenceUploadedEvent(
+                caseData,
+                buildCaseDataWithConfidentialDocuments(REP_USER),
+                false,
+                userDetailsRespondentSolicitor()
+            );
+
+        when(furtherEvidenceNotificationService.getRespondentRepresentativeEmails(caseData)).thenReturn(
+            representativeSolicitors);
+        when(furtherEvidenceNotificationService.getLocalAuthoritySolicitorEmails(caseData)).thenReturn(laSolicitors);
+
+        furtherEvidenceUploadedEventHandler.handleDocumentUploadedEvent(furtherEvidenceUploadedEvent);
+
+        verify(furtherEvidenceNotificationService).sendFurtherEvidenceDocumentsUploadedNotification(
+            caseData,
+            representativeSolicitors.stream()
```

Comment:
this does not looks right, make is as simple as it can be:
Set.of("rep_solicitor1@example.com", "rep_solicitor2@example.com"),

Reply:
Fixed